packages/build/src/homebrew/utils.ts

```diff
@@ -1,13 +1,68 @@
 import crypto from 'crypto';
 import https from 'https';
 
-export function httpsSha256(url: string): Promise<string> {
-  return new Promise((resolve, reject) => {
+export async function npmPackageSha256(
+  packageUrl: string,
+  httpGetFn: typeof httpGet = httpGet
+): Promise<string> {
+  const json = await httpGetFn(packageUrl, 'json');
+  const tarballUrl = json.dist.tarball;
+  const shasum = json.dist.shasum;
+
+  const tarball = await getTarballWithRetries(tarballUrl, shasum, httpGetFn);
+  const hash = crypto.createHash('sha256');
+  hash.update(tarball);
+  return hash.digest('hex');
+}
+
+async function getTarballWithRetries(
+  url: string,
+  shasum: string,
+  httpGetFn: typeof httpGet,
+  attempts = 3
+): Promise<Buffer> {
+  try {
+    const tarball = await httpGetFn(url, 'binary');
+    const hash = crypto.createHash('sha1').update(tarball).digest('hex');
+    if (hash !== shasum) {
+      throw new Error(`shasum mismatch: expected '${shasum}', got '${hash}'`);
+    }
+
+    return tarball;
+  } catch (err) {
+    if (attempts === 0) {
+      throw err;
+    }
+
+    return getTarballWithRetries(url, shasum, httpGetFn, attempts - 1);
+  }
+}
+
+export function httpGet(url: string, response: 'json'): Promise<any>;
+export function httpGet(url: string, response: 'binary'): Promise<Buffer>;
+
+export async function httpGet(
+  url: string,
+  responseType: 'json' | 'binary'
+): Promise<any | Buffer> {
+  const response = await new Promise<string | Buffer[]>((resolve, reject) => {
     https.get(url, (stream) => {
-      const hash = crypto.createHash('sha256');
+      let data: string | Buffer[] = responseType === 'json' ? '' : [];
       stream.on('error', (err) => reject(err));
-      stream.on('data', (chunk) => hash.update(chunk));
-      stream.on('end', () => resolve(hash.digest('hex')));
+      stream.on('data', (chunk) => {
+        if (typeof data === 'string') {
+          data += chunk;
```

Comment:
You'll want to call `stream.setEncoding('utf8');` in this case in order to properly handle multibyte characters.